src/language-server.ts

```diff
@@ -14,10 +14,23 @@ import ReleaseVersion from "./release/version";
 import extract = require("extract-zip");
 
 namespace LanguageServer {
+	function isExecutableFile(path: fs.PathLike): boolean {
+		const stat = fs.lstatSync(path);
+		let access = false;
+		try {
+			fs.accessSync(path, fs.constants.R_OK);
```

Comment:
```suggestion
			fs.accessSync(path, fs.constants.X_OK);
```
Unless there's something I don't understand, I think we want to check for execute permission, not read?

Also, `undefined | throw` is a very interesting way of saying `boolean`, good job Node :+1:

Reply:
I was shocked, shocked to see this approach. 
Also, why `R_OK` rather than `READ` and why name it `constants` rather than `permissions`
this could have been
```js
fs.hasAccess(path,  fs.permissions.READ | fs.permissions.WRITE)
```